lang/functions_test.go

```diff
@@ -1044,6 +1057,12 @@ func TestFunctions(t *testing.T) {
 		Data:    data,
 		BaseDir: "./testdata/functions-test", // for the functions that read from the filesystem
 	}
+	// We may need to activate some experiments if some of our functions
+	// are currently experimental. This set should be empty if there are no
+	// active experiments involving functions.
+	experimentsSet := experiments.NewSet()
+	experimentsSet.Add(experiments.ModuleVariableOptionalAttrs) // for the experimental "default" function
+	scope.SetActiveExperiments(experimentsSet)
```

Comment:
Can we test the experiment not-enabled case?

Reply:
With the experiments mechanism as it is currently designed, I didn't really see any practical way to do that without considerable refactoring. With fresh eyes though, I have a new idea which I will try before moving forward with this.

Reply:
I reworked this test to have more explicit awareness of the idea of an experimental function, so we can now test both ways.